### PR TITLE
add consumer key helpers

### PR DESCRIPTION
**Pull request by yadutaf**
Following discussion with @VincentCasse on php-ovh-sms, here is a design proposal for improved consumer key helper. See Readme and test for examples. But, in a nutshell, it allows to build autorization requests for a given api endpoint prefix and and all sub-path or only the specific route.

ReadOnly is mapped to "GET". ReadWrite to all HTTP Methods.

This PR is about the design itself. If we all agree, we'll implement this in the other wrappers.

At a later time, we may consider human readable aliases for sections of /me and /auth like
- "identity"
- "billing"
- "credentials"
...

ping @gregdel, @VincentCasse

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -161,10 +161,10 @@ func main() {
 	ckReq := client.NewCkRequest()
 
 	// Allow GET method on /me
-	ckReq.AddRule("GET", "/me")
+	ckReq.GenerateRulesSingle("/xdsl", ovh.ReadOnly)
```
Comment (PouuleT) [inline README.md]: Typo, xdsl -> me

File: ovh/consumer_key.go [hunk 2]
```diff
@@ -46,18 +62,60 @@ func (ck *CkValidationState) String() string {
...
 // AddRule adds a new rule to the ckRequest
 func (ck *CkRequest) AddRule(method, path string) {
-	ck.AccessRules = append(ck.AccessRules, &AccessRule{
+	ck.AccessRules = append(ck.AccessRules, AccessRule{
 		Method: method,
 		Path:   path,
 	})
 }
 
+// GenerateRules generates alls rules for a given API endpoint in a human friendlier
```
Comment (PouuleT) [inline ovh/consumer_key.go]: Typo *alls =)

**Comment by yadutaf** [comment]
Adding @LukeMarlin to the discussion for C# wrapper

**Comment by LukeMarlin** [comment]
Cool feature. Only problem for me is that it can be too restrictive (RW contains too much). I'd add an overloaded method taking custom verbs as a `String[]`

**Comment by gregdel** [comment]
It's only helpers, the goal is to make the developer life easier. There is still a "AddRule" method for fine-grained right control.

**Comment by LukeMarlin** [comment]
Sure, but AddRule does not support the nice recurse param.

**Comment by yadutaf** [comment]
> Only problem for me is that it can be too restrictive (RW contains too much)
Would it be better with like an intermediate ``ReadWriteDelete``?

**Comment by VincentCasse** [comment]
yep, distinct delete could ba a good thing.. Maybe distinct write and others actions ? but i don't know how

**Comment by yadutaf** [comment]
Adding ``ReadWriteSafe`` without the Delete

File: ovh/consumer_key.go (hunk 2 above)
Comment (yadutaf) [inline ovh/consumer_key.go]: Fixes, thx.

File: README.md (hunk 1 above)
Comment (yadutaf) [inline README.md]: Fixed. Thx.

**Comment by yadutaf** [comment]
v2 pushed with Typo fixes from @gregdel  and @PouuleT and intermediate authorization level suggested by @LukeMarlin.

**Comment by PouuleT** [comment]
Why not do something like
AddRules([]string{"GET","POST","PUT"}, "/sms")
and
AddRecursiveRules([]string{"GET","POST","PUT", "/sms")

**Comment by yadutaf** [comment]
API could be like:

```go
const ( 
    ReadOnly      = []string{"GET"}
    ReadWrite     = []string{"GET", "POST", "PUT", "DELETE"}
    ReadWriteSafe = []string{"GET", "POST", "PUT"}
)
    
func AddRule(verb, pattern string)                  {}
func AddRules(verbs []string, pattern string)       {}    
func AddRecursiveRules(verbs []string, path string) {}
```

Go ?

**Comment by LukeMarlin** [comment]
Exactly, +1

**Comment by yadutaf** [comment]
@cygy does is look good to you too for Swift wrapper ?

**Comment by cygy** [comment]
I have already some helpers but I can update if needed.
Seems good.

**Comment by yadutaf** [comment]
And there it is. I much prefer this new API anyway. Less code, less magic. As far as I'm concerned, we're ready to go.

File: README.md [hunk 3]
```diff
@@ -161,10 +161,10 @@ func main() {
 	ckReq := client.NewCkRequest()
 
 	// Allow GET method on /me
-	ckReq.AddRule("GET", "/me")
+	ckReq.AddRules(ovh.ReadOnly, "/me")
 
 	// Allow GET method on /xdsl and all its sub routes
-	ckReq.AddRule("GET", "/xdsl/*")
+	ckReq.AddRules(ovh.ReadOnly, "/xdsl")
```
Comment (gregdel) [inline README.md]: This should be:
```go
ckReq.AddRecursiveRules(ovh.ReadOnly, "/xdsl")
```
To match the comment above

Comment (yadutaf) [inline README.md]: Damn it. Ur right.

Comment (yadutaf) [inline README.md]: Fixed.

**Comment by yadutaf** [comment]
As far as I'm concerned: Ready.

**Comment by gregdel** [comment]
Merged, thanks @yadutaf. Good work !

**Comment by yadutaf** [comment]
\o/ Starting Python version.

**Pull request merged**